signatures.md

```diff
@@ -52,163 +49,350 @@ Note: `Message` is used here as the object being signed, but this interface shou
-        // particular signature. In the case of ECDSA signatures, this function can 
-        // be fulfilled via the 'ECRecover' method. If a different signature scheme 
-        // is used, then some other mechanism of 'recovering' a message authors 
-        // public key must be provided.
+        // Recover determines the public key associated with a particular signature. For
+  			// ECDSA signatures public keys can be recovered from SignatureBytes and an
+  			// associated message. For BLS signatures, public keys will be recovered by lookup 
+  			// using a mapping from Address to public key.
         //
         // Out:
-        //    pk - the public key associated with `M` who signed `m`
+        //    pk - the public key associated with `M` who signed `msg`
         //    err - a standard error message indicating any process issues
         //    **
         // In:
-        //    m - a series of bytes representing the signed message
-        //    sig - a series of bytes representing a signature usually `r`|`s`
+        //    msg - a series of bytes representing the signed message
+        //    sig - a series of bytes representing a signature
         //
-        Recover(m Message, sig SignatureBytes) (pk PublicKey, err error)
+        Recover(msg Message, addr Address, sig SignatureBytes) (pk PublicKey, err error)
 }
 ```
 
+### Signed Message
 
-# Selected Signature Scheme
+This is the container for a signature that combines a message with an address and signature. 
+
+```go
+Type SignedMessage Struct {
 
-Currently, Filecoin uses secp256k1 signatures to fulfill the above interface. All signatures on messages, blocks, and tickets currently use the same scheme and format.
+		// Addrs is an array of Addresses associated with a particular message in Msgs 
+  	// at the same index. Each Address tracks the ID of the actor responsible for 
+  	// signing a particular Msg. An ID can be converted into a public key via 
+  	// Address.GetPublicKey(ID).
+		//
+		Addrs []Address `JSON`
+
+  	// Msgs is an array of messages. Each message in Msgs is represented by raw bytes.
+  	// Each message in Msgs will correspond to an Address in Addrs by index. 
+  	//
+    Msgs []bytes `JSON`   
+
+  	// Sig is a cryptographic signature that was generated according to the type 
+  	// reflected in Address.
+    Sig SignatureBytes `JSON` 
+    
+}
+```
 
-## Wire Format
+## Key Generation & Elliptic Curve Parameters
 
-What bits are on the wire and in what order/format. We are currently adhering to libsecp256k1 serialization which is laid out as follows. Note that this format description may not be accurate. See the github link below for an authoritative format description.
+This specification assumes the implementer uses a strong cryptographic pseudorandom generator `RANDOM`. The following was adapted from [IETF BLS Signature Scheme - Keygen](https://tools.ietf.org/html/draft-boneh-bls-signature-00#section-2.2). 
```

Comment:
do you have a good link for a definition of a strong cryptographic pseudorandom [number] generator?

Reply:
Will add link to [NIST](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf). Incidentally, the authors Elaine Barker and John Kelsey are fairly responsive if we have further questions or need clarifications.

Reply:
Maybe also https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator which is likely easier to understand from a high level persepective

Reply:
Will add link in References section